.github/workflows/pr-build.yaml

```diff
@@ -0,0 +1,61 @@
+name: PR Build
+
+on:
+  pull_request:
+    branches: [main]
+env:
+  REGISTERY: ghcr.io
+  NAMESPACE: galasa-dev
+  
+jobs:
+    Build-using-openapi2beans-and-gradle:
+        name: Build framework using openapi2beans and gradle
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Code
+              uses: actions/checkout@v4
+            
+            - name: Setup JDK 
+              uses: actions/setup-java@v4
+              with:
+                java-version: '11'
+                distribution: 'semeru'
+            
+            - name: Print githash
+              run: |
+                echo $GITHUB_SHA > ./framework.githash
+              
+            - name: Setup Gradle
+              uses: gradle/actions/setup-gradle@v3
+              with:
+                gradle-version: 6.9.2
+            
+            - name: Build openapi2beans
+              run: |
+               docker run --rm -v ${{ github.workspace }}:/var/workspace ghcr.io/jaydee029/openapi2beans:main generate --yaml var/workspace/${{ env.YAML_LOCATION }} --output var/workspace/${{ env.OUTPUT_LOCATION }} --package ${{ env.PACKAGE }}
+
+              env:
+                YAML_LOCATION: "galasa-parent/dev.galasa.framework.api.openapi/src/main/resources/openapi.yaml"  # Update this path
+                OUTPUT_LOCATION: "generated-beans"
+                PACKAGE: "generated.beans"
+            
+            - name: Test and Build Framework
+              run: |
+               gradle -b galasa-parent/build.gradle check publish \
+               -Dorg.gradle.jvmargs=-Xmx5120M \
+               -PsourceMaven=https://repo.maven.apache.org/maven2/  \
+               -PcentralMaven=https://repo.maven.apache.org/maven2/ \
+               -PtargetMaven=${{ github.workspace }}/repo
+               
+            - name: Build framework Image for testing
+              uses: docker/build-push-action@v5
+              with:
+                context: .
+                file: dockerfiles/dockerfile.framework
+                load: true
+                tags: framework:test

```

Comment:
Can we change this to `framework:<PR NUMBER>` so multiple PRs don't overwrite the same image?